pinot-core/src/main/java/org/apache/pinot/core/operator/filter/predicate/RegexpLikePredicateEvaluatorFactory.java

```diff
@@ -36,21 +39,33 @@ public class RegexpLikePredicateEvaluatorFactory {
   private RegexpLikePredicateEvaluatorFactory() {
   }
 
-  /// When the cardinality of the dictionary is less than this threshold, scan the dictionary to get the matching ids.
-  public static final int DICTIONARY_CARDINALITY_THRESHOLD_FOR_SCAN = 10000;
+  /// Default threshold when the cardinality of the dictionary is less than this threshold,
+  // scan the dictionary to get the matching ids.
+  public static final int DEFAULT_DICTIONARY_CARDINALITY_THRESHOLD_FOR_SCAN = 10000;
 
   /**
-   * Create a new instance of dictionary based REGEXP_LIKE predicate evaluator.
+   * Create a new instance of dictionary based REGEXP_LIKE predicate evaluator with configurable threshold.
    *
    * @param regexpLikePredicate REGEXP_LIKE predicate to evaluate
-   * @param dictionary Dictionary for the column
-   * @param dataType Data type for the column
+   * @param dictionary          Dictionary for the column
+   * @param dataType            Data type for the column
+   * @param numDocs             Number of documents in the segment
+   * @param queryContext        Query context containing query options (can be null)
    * @return Dictionary based REGEXP_LIKE predicate evaluator
    */
   public static BaseDictionaryBasedPredicateEvaluator newDictionaryBasedEvaluator(
-      RegexpLikePredicate regexpLikePredicate, Dictionary dictionary, DataType dataType) {
+      RegexpLikePredicate regexpLikePredicate, Dictionary dictionary, DataType dataType, int numDocs,
+      QueryContext queryContext) {
```

Comment:
Can you create another static factory method with the same signature as the previous one? Probably nobody outside this repo uses this method, but it is a good practice not to break binary compatibility